src/Symfony/Component/Security/Http/EventListener/SessionStrategyListener.php

```diff
@@ -39,7 +39,7 @@ public function onSuccessfulLogin(LoginSuccessEvent $event): void
         $request = $event->getRequest();
         $token = $event->getAuthenticatedToken();
 
-        if (!$request->hasSession() || !$request->hasPreviousSession()) {
+        if (!$request->hasSession() || !$request->hasPreviousSession() || $event->getPreviousToken()->getUser() === ($token->getUser() ?? false)) {
```

Comment:
should we check the identity of the user instance or should we check the user identifier ?

A previous token might contain an unserialized user, which could be a different PHP object instance than the object returned by the user provider, while still being the same authenticated user from a business point of view.

Reply:
At this point, the token has been refreshed by ContextListener. The situation you describe might exist though, I'm no expert here. My proposal it to start this way and see later if we need more fuzzy matching (what ContextListener already implements.)